.travis.yml

```diff
@@ -1,106 +1,34 @@
-sudo: false
+# We will use docker to set up our environment, so don't use any particular
+# language in Travis itself
+language: generic
 
-language: cpp
+# Enable docker
+sudo: required
+services:
+    - docker
 
-os:
-  # - linux
-  - osx
+# Disable automatic submodule fetching ...
+git:
+    submodules: false
 
-osx_image: xcode8
+# ... in order to ensure we do a non-shallow submodule fetch
+before_install: git submodule update --init --no-recommend-shallow
 
 env:
-  # code analysis
-  # - TASK=lint
-  # build mxnet.so with CUDA
-  # - TASK=build
-  # run tests/cpp
-  - TASK=cpp_test
-  # run tests/python
-  - TASK=python_test
-  - TASK=r_test
-  # - TASK=julia JULIA_VER=0.4
-  # - TASK=scala_test
-
-  # TODO, R test, distributed test, clang, more g++ versions
-
-matrix:
-  include:
-    - # os: linux
-      # dist: trusty
-      # env: TASK=perl_test
-    - os: osx
-      ## sudo is required because
-      ## prexexisting packages conflict
-      ## with new ones.
-      ## would be nice to have macports
-      ## on travis osx, it has all needed perl packages
-      sudo: required
-      env: TASK=perl_test
-#       env: TASK=julia JULIA_VER=0.4
-#     - os: linux
-#       env: TASK=build
-#     - os: linux
-#       env: TASK=cpp_test
-#     - os: linux
-#       env: TASK=python_test
-#     - os: linux
-#       env: TASK=r_test
-#     - os: linux
-#       env: TASK=scala_test
-
-# dependent apt packages
-addons:
-  apt:
-    sources:
-      - ubuntu-toolchain-r-test
-    packages:
-      - doxygen
-      - wget
-      - git
-      - libcurl4-openssl-dev
-      - unzip
-      - libatlas-dev
-      - libopencv-dev
-      - gcc-4.8
-      - g++-4.8
-      - python-numpy
-      - python-nose
-      - python3-numpy
-      - python3-dev
-      - python3-nose
-      - graphviz
-      - libmouse-perl
-      - pdl
-      - cpanminus
-      - swig
-      - libgraphviz-perl
```

Comment:
@leandro-lucarella-sociomantic no, we don't need those dependencies.  All of this stuff is for things not in the underlying C library.  How do you think we were able to build `libmxnet` all this time without them? ;-)